### PR TITLE
make the projects page responsive

### PR DESCRIPTION
**Pull request by dollarbeale**
### **Before**

image
image

### **After**
image
image

### **The Breathe Button has also been adjusted for mobile:**

Before
image

After
image

### CODE REVIEW

File: src/pages/projects.jsx [hunk 1]
```diff
@@ -1,109 +1,154 @@
...
+            margin-bottom: 80px;
+            line-height: 1.5;
+
+            @media screen and (max-width: 900px), (max-height: 640px) {
+                font-size: 25px;
+            }
+        `}
+
+    ${(props) =>
+        props.project &&
```
Comment (ssaberan) [inline src/pages/projects.jsx]: Instead of having two props, `main` and `project`, why not just have one prop, and if it's `true` do one thing, otherwise (so if it's `false`) do the other thing? It makes the code cleaner and reduces complexity.

In general, if you want to implement logic that follows 2 possibilities, you should only use 1 boolean, not 2, because a boolean holds 2 states. (So if you're using 2 booleans, there should be 4 possibilities you want)

**Comment by ssaberan** [comment]
Can you rename the commit message from `make the projects page reflexive` to `make the projects page responsive`?

File: src/pages/projects.jsx (hunk 1 above)
Comment (dollarbeale) [reply src/pages/projects.jsx]: Hi Soroush! I actually plan to continue changing my projects page to have more header and paragraph variations, so I feel like it might be better to not use a boolean, as I would have to undo it in the next edit anyway. What are your thoughts?

Comment (ssaberan) [reply src/pages/projects.jsx]: I'm saying you're currently using 2 booleans when you only need 1.

It sounds like you're saying you want to use a string instead of a boolean since you plan to have multiple options in the future, which is fine, but just don't use 2 booleans when you only need 1 like you're doing in this PR.

Comment (dollarbeale) [reply src/pages/projects.jsx]: Sorry for the trouble, but I am having a hard time understanding. Are you saying I should only use the line `@media screen and (max-width: 900px), (max-height: 640px)` once per styled component (i.e. it is written only once and covers both props)? I've been reading the styled-components documentation, but I still am not understanding sadly.

Comment (ssaberan) [reply src/pages/projects.jsx]: Right now you're using a short-circuit (the `&&`) to render:
```
css`
    font-size: 40px;
    margin-bottom: 80px;
    line-height: 1.5;
    @media screen and (max-width: 900px), (max-height: 640px) {
        font-size: 25px;
    }
`
```
if `props.main` is true, and to render nothing if `props.main` is false. Let's call this snippet `A`. Then in the same styled component, you're using another short-circuit to render:
```
css`
    font-size: 30px;
    margin-bottom: 20px;
    line-height: 1.5;
    @media screen and (max-width: 900px), (max-height: 640px) {
        font-size: 20px;
    }
`
```
if `props.project` is true, and to render nothing if `props.project` is false. Let's call this snippet `B`. Instead, you should render the css for `props.main` when `props.main` is true, and render the css for `props.project` when `props.main` is false. One way you can do this is by using a ternary instead of a short-circuit: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

So in summary, what you're currently doing:
Using a short-circuit to render `A` or nothing
Then using a short-circuit to render `B` or nothing

What you should be doing:
Using a ternary to render `A` or `B`

Also note that your `Header` styled component has this same issue.

Comment (ssaberan) [reply src/pages/projects.jsx]: But if you want to use a string instead of a boolean for `props.main`, instead of a ternary you can use a switch statement: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

In which case I think a name like `props.layoutType` or something might be better. Then you can use `props.layoutType` as your switch expression, and then choose what css to render in the case statements.

Comment (ssaberan) [reply src/pages/projects.jsx]: But now that I think of it, if your styled component solely uses one case of css out of many, it makes a lot more sense to just have a separate styled component for each case of css. Having branching logic to decide what the entire value is for the styled component doesn't make any sense when you could just make a separate styled component instead.

Comment (dollarbeale) [reply src/pages/projects.jsx]: I think I see what you're saying! Thank you very much for explaining and showing the alternatives! I decided to separate them all into styled components to make it cleaner.

**Review by ssaberan — APPROVED** [review]
Nice job!

**Comment by dollarbeale** [comment]
> Nice job!

Thank you!
